editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.tsx

```diff
@@ -10,17 +11,26 @@ import QuestionHeader from "../shared/Preview/QuestionHeader";
 import { Dropzone } from "../shared/PrivateFileUpload/Dropzone";
 import { FileStatus } from "../shared/PrivateFileUpload/FileStatus";
 import { UploadedFileCard } from "../shared/PrivateFileUpload/UploadedFileCard";
-import { MultipleFileUpload } from "./model";
+import { createFileList, FileList, MultipleFileUpload } from "./model";
 
 type Props = PublicProps<MultipleFileUpload>;
 
-export default Component;
-
 function Component(props: Props) {
   const [slots, setSlots] = useState<FileUploadSlot[]>([]);
   const [fileUploadStatus, setFileUploadStatus] = useState<string | undefined>(
     undefined
   );
+  const [fileList, setFileList] = useState<FileList>({
+    required: [],
+    recommended: [],
+    optional: [],
+  });
```

Comment:
This is the data structure which _I think_ can drive this component...
 - It can display the list of files on right of the initial page
 - It can populate the modal
 - It can map a slot to a `FileType` instance for parsing on `handleSubmit()`

Reply:
:100: I think this also covers everything!

#1751 introduces a `slotsSchema` and local `handleSubmit()` to get us started around your last point